syntax = "proto3";
option go_package = "./grpc_comms";

// The greeting service definition.
service VolpeContainer {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc InitFromSeed (Seed) returns (Reply) {}

  rpc InitFromSeedPopulation (Population) returns (Reply) {}

  rpc GetBestPopulation (PopulationSize) returns (Population) {}

  rpc AdjustPopulationSize (PopulationSize) returns (Reply) {}

  rpc RunForGenerations (PopulationSize) returns (Reply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

message Seed {
  int32 seed = 1;
}

message PopulationSize {
  int32 size = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}


message Individual {
  bytes genotype = 1;
  float fitness = 2;
}

message Population {
  repeated Individual members = 1;
}

message Reply {
  bool success = 1;
  optional string message = 2;
}